citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -438,6 +439,29 @@ public CitationPage getPrevAndNextCitation(IUser user, String groupId, String co
         }
         return result;
     }
+
+    @Override
+    public List<ItemDeletionResponse> deleteCitations(IUser user, String groupId, List<String> citationIdList)
+            throws ZoteroConnectionException, ZoteroHttpStatusException {
+        List<ItemDeletionResponse> response = zoteroManager.deleteMultipleItems(user, groupId, citationIdList,
+                zoteroManager.getLatestGroupVersion(user, groupId));
+        int counter = 0;
+        for (ItemDeletionResponse res : response) {
+            if (res.equals(ItemDeletionResponse.SUCCESS)) {
+                for (int i = 0; i < 50; i++) {
```

Comment:
see my comment on the plugin pr. Once we have the ids for each response, this should loop over the ids rather than over 50. If 50 ever changes, we otherwise have to change not just the plugin but this code as well.